homeassistant/components/deconz.py

```diff
@@ -0,0 +1,198 @@
+ATTR_MODEL_ID = 'model_number'
+ATTR_UNIQUE_ID = 'uniqueid'
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup services for Deconz."""
+    deconz_config = config[DOMAIN]
+    config_file = load_json(hass.config.path(CONFIG_FILE))
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))[DOMAIN]
+
+    @callback
+    def _shutdown(call):  # pylint: disable=unused-argument
+        """Stop the connections to Deconz on shutdown."""
+        if DECONZ_DATA in hass.data:
+            _LOGGER.info("Stopping Deconz session.")
+            hass.data[DECONZ_DATA].close()
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _shutdown)
+
+    @asyncio.coroutine
+    def _configure(call):
+        """Set attribute of device in Deconz.
+
+        Field is a string representing a specific device in Deconz
+        e.g. field='/lights/1/state'.
+        Data is a json object with what data you want to alter
+        e.g. data={'on': true}.
+        {
+            "field": "/lights/1/state",
+            "data": {"on": true}
+        }
+        See Dresden Elektroniks REST API documentation for details:
+        http://dresden-elektronik.github.io/deconz-rest-doc/rest/
+        """
+        deconz = hass.data[DECONZ_DATA]
+        field = call.data.get('field')
+        data = call.data.get('data')
+        yield from deconz.put_state_async(field, data)
+    hass.services.async_register(
+        DOMAIN, 'configure', _configure, descriptions['configure'])
+
+    if CONF_API_KEY in deconz_config:
+        pass
+    elif CONF_API_KEY in config_file:
+        deconz_config[CONF_API_KEY] = config_file[CONF_API_KEY]
+    else:
+        from pydeconz.utils import get_api_key
+        api_key = yield from get_api_key(hass.loop, **deconz_config)
+        if api_key:
+            deconz_config[CONF_API_KEY] = api_key
+            save_json(hass.config.path(CONFIG_FILE), {CONF_API_KEY: api_key})
+        else:
+            hass.async_add_job(
+                request_configuration, hass, config, deconz_config)
+            return True
+    result = yield from _setup_deconz(hass, config, deconz_config)
+    return result is not False
+
+
+@asyncio.coroutine
+def _setup_deconz(hass, config, deconz_config):
+    """Setup Deconz session.
+
+    Load config data for server information.
+    Load group data containing which light groups are available.
+    Load light data containing which lights are available.
+    Load sensor data containing which sensors are available.
+    Start websocket for push notification of state changes from Deconz.
+    """
+    from pydeconz import DeconzSession
+    deconz = DeconzSession(hass.loop, **deconz_config)
+    hass.data[DECONZ_DATA] = deconz
+    result = yield from deconz.load_parameters()
+    if result is False:
+        _LOGGER.error('Failed to setup deCONZ component')
+        return False
+    hass.async_add_job(discovery.async_load_platform(hass,
```

Comment:
You can satisfy linting by just adding an enter after `(` and indent the parameter list with 4 spaces. Better readable.

Reply:
Done